### DISCUSSION TITLE
[FEATURE REQUEST] Update only a specific branch with fileserver.update

### DISCUSSION THREAD

**Discussion by FredrikLiindgren** [saltstack/salt#67507] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
I'm frustrated when I push custom module/state/runner changes to a branch and have to update the entire remote if I want to speed up the synchronization of changes.

**Describe the solution you'd like**
I want to be able to just update a single branch on a remote with `fileserver.update`.
https://github.com/saltstack/salt/blob/a686ce00b240d97410fed9f0d76e74c218da116c/salt/runners/fileserver.py#L322
Something like this: `salt-run fileserver.update backend=gitfs remote=my-remote branch=my-branch`

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://via.vmw.com/salt-slack)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by OrangeDog** [comment]
That's not really going to improve anything. It's very unlikely that you have so many changes on other branches that there will be a noticeable difference. Just typing the option will probably take longer.
